appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/BlocklyPanel.java

```diff
@@ -808,6 +826,38 @@ public static String getLocalizedComponentType(String key) {
     return TranslationDesignerPallete.getCorrespondingString(key);
   }
 
+  public static void switchToBlocksEditor(){
+    Ode.getInstance().getDesignToolbar().switchToBlocksEditor();
+  }
+
+  public static void switchToFormEditor(){
+    Ode.getInstance().getDesignToolbar().switchToFormEditor();
+  }
+  public static void displayDialog(){
+    Walkthrough.displayDialog();
+  }
+  public static String getProjectId(){
+    return String.valueOf(Ode.getInstance().getCurrentYoungAndroidProjectRootNode().getProjectId());
+  }
+  public static boolean inBlocksView(){
+    return Ode.getInstance().getDesignToolbar().inBlocksView();
+  }
+  public static ArrayList<String> getComponentNames(){
+    String currentScreen =  Ode.getInstance().getDesignToolbar().getCurrentProject().currentScreen;
+    
+    ArrayList<String> ComponentNames = new ArrayList<String>();
+    ProjectEditor projectEditor = Ode.getInstance().getDesignToolbar().getCurrentProject().screens.get(currentScreen).formEditor.getProjectEditor();
+    if (projectEditor instanceof YaProjectEditor) {
+        YaProjectEditor yaProjectEditor = (YaProjectEditor) projectEditor;
```

Comment:
What should it be? Is the standard 2 spaces or 3 or 4? This looks consistent to me, so I must be missing something.